### PR TITLE
[nng] update nng to the latest version v1.2.5

### PR DESCRIPTION
**Pull request by ggessner**
update the nng library to the latest tagged version in order to work correctly with nngpp.

### CODE REVIEW

File: ports/nng/portfile.cmake [hunk 1]
```diff
@@ -3,8 +3,8 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
```
Comment (LilyWangL) [inline ports/nng/portfile.cmake]: `include(vcpkg_common_functions)` is deprecated. Please remove it.

File: ports/nng/CONTROL [hunk 2]
```diff
@@ -1,5 +1,5 @@
 Source: nng
-Version: 2019-02-27
+Version: v1.2.5
 Description: NNG, like its predecessors nanomsg (and to some extent ZeroMQ), is a lightweight, broker-less library, offering a simple API to solve common recurring messaging problems, such as publish/subscribe, RPC-style request/reply, or service discovery.
```
Comment (LilyWangL) [inline ports/nng/CONTROL]: Can you please add the homepage for this port in the CONTROL file?
```
Homepage: https://github.com/nanomsg/nng
```

**Review by LilyWangL — CHANGES_REQUESTED** [review]

**Comment by LilyWangL** [comment]
@ggessner Can you please use command `file(INSTALL ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright?

File: ports/nng/portfile.cmake (hunk 1 above)
Comment (ggessner) [reply ports/nng/portfile.cmake]: removed the deprecated command.

File: ports/nng/CONTROL (hunk 2 above)
Comment (ggessner) [reply ports/nng/CONTROL]: added homepage.
Thank you

**Review by LilyWangL — APPROVED** [review]

File: ports/nng/CONTROL [hunk 3]
```diff
@@ -1,6 +1,7 @@
 Source: nng
-Version: 2019-02-27
+Version: v1.2.5
```
Comment (ras0219-msft) [inline ports/nng/CONTROL]: Conventionally, we prettify this version field to just `1.2.5`.

File: ports/nng/portfile.cmake [hunk 4]
```diff
@@ -1,10 +1,8 @@
-include(vcpkg_common_functions)
-
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO nanomsg/nng
-    REF ddeac617c9914284038241870cb99ae174fb3755
-    SHA512 c45f322dfa3fba42db0561d95546159cc5e0768345d27f0bee53bdb71e77d5892e2e9bea50a625094ebf3c67b0c3fe0a8edea2660ebcb4fd0991fb0602055bc1
+    REF v1.2.5
```
Comment (ras0219-msft) [inline ports/nng/portfile.cmake]: We prefer to use the SHA1 of the tag here, so that it cannot change.

File: ports/nng/portfile.cmake [hunk 5]
```diff
@@ -55,6 +53,6 @@ else()
 endif()
 
 # Put the licence file where vcpkg expects it
-configure_file(${SOURCE_PATH}/LICENSE.txt ${CURRENT_PACKAGES_DIR}/share/nng/copyright COPYONLY)
+file(INSTALL ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (ras0219-msft) [inline ports/nng/portfile.cmake]: The `configure_file()` command is correct here, please revert this change.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
LGTM once my three comments are addressed.

**Comment by ras0219-msft** [comment]
Thanks for addressing all my comments!
